### PR TITLE
Prevent sending invites to already available assignee/admin

### PR DESCRIPTION
**Pull request by mojo706**
#### What does this PR do?
- Add a check for whether a user already exists and stops them from being re-invited. 
#### How should this be manually tested?
- Clone this branch
- On Postman try to invite an existing `assignee` or `admin` 
- You should get a message informing you that you cannot invite an existing user
#### What are the relevant pivotal tracker stories?
[#160482024](https://www.pivotaltracker.com/story/show/160482024)
#### Screenshots (if appropriate)
N/A

### CODE REVIEW

File: src/server/controllers/users.js [hunk 1]
```diff
@@ -150,13 +150,7 @@ module.exports = {
...
         .spread( async (createdUser, created) => {
           if(!created) {
-            await User.update(req.body,{
-              where: {
-                email: userObject.email
-              },
-              returning: true
-            });
-            return res.status(200).send({ message: 'the user role has been updated' });
+            return res.status(400).send({ message: 'The user with that email address already exists' });
```
Comment (kimobrian) [inline src/server/controllers/users.js]: @mojo706 Update this functionality to cater for a user who was created thru creating an incident. It means their role ID is still for a regular user. So within check that the role ID matches the role ID on the invite, if not update their role Id and send them the invite email. If they exist and their roleId marches the roleId on the invite, then reply with the above message: `The user with that email address already exists as an {assignee/admin}. Try updating their role`

**Review by kimobrian — CHANGES_REQUESTED** [review]
Check comments.

File: src/server/controllers/users.js (hunk 1 above)
Comment (mojo706) [reply src/server/controllers/users.js]: okay

File: src/server/controllers/users.js [hunk 2]
```diff
@@ -149,14 +149,19 @@ module.exports = {
...
           if(!created) {
-            await User.update(req.body,{
-              where: {
-                email: userObject.email
-              },
-              returning: true
-            });
-            return res.status(200).send({ message: 'the user role has been updated' });
+            if(userExists.roleId === req.body.roleId) {
+              return res.status(400).send({ message: 'The user with that email address already exists as an {assignee/admin}. Try updating their role' });
```
Comment (kimobrian) [inline src/server/controllers/users.js]: Can we get the actual role here @mojo706 . Instead of both names like `assignee/admin`.

Comment (mojo706) [reply src/server/controllers/users.js]: Sure thing

File: src/server/controllers/users.js [hunk 3]
```diff
@@ -149,14 +152,14 @@ module.exports = {
...
-            await User.update(req.body,{
-              where: {
-                email: userObject.email
-              },
-              returning: true
-            });
-            return res.status(200).send({ message: 'the user role has been updated' });
+            if(userExists.roleId === req.body.roleId) {
+              const roleName = await matchRoleIdToRoleName(req.body.roleId);
+              return res.status(400).send({ message: `The user with that email address already exists as an ${roleName} . Try updating their role` });
```
Comment (kimobrian) [inline src/server/controllers/users.js:159]: Seriously, this is a new user we are inviting, we can't use the role in the body. For example if we're inviting the user as an assignee and maybe they are in the DB as admin, they should be notified that they exist as an admin. Your logic will say they exist as assignee.

**Pull request closed without merging**